### PR TITLE
happy honk meal box dispenser

### PR DESCRIPTION
**Pull request by brainfood1183**
untitled-149

Happy Honk Meals

- Adds happy honk meal dispenser
- Adds happy honk clown meal box
- Adds happy honk mime meal box
- Adds woeful cluwne meal box(emag only)
- Adds robust nukie meal box(emag only)
- all meal boxes come with a random toy.
- No food is present in any of the boxes and chefs should put food in them before giving them to crewmembers.
- The average value using an appraisal tool for the happy honk dispenser and contents is 1200.
- Adds golden honker bicycle horn, there is a low low chance for it to spawn in a happy honk meal box. The horn can be used to pray and admins can reward the lucky crewmember with something in exchange for the golden honker.
- There is a low chance for c4 to spawn in the robust nukie meal box (emag only). 
- There is a refill box and refill crate which is orderable from cargo.
- The happy honk meal dispenser is locked to kitchen access only.

Mappers should either place a happy honk meal dispenser in the kitchen or in maintenance if they would like one on there map. The main purpose of happy honk meal dispenser is rp and hugs.

### CODE REVIEW

**Comment by Jackrost** [comment]
Mc Honkals

File: Resources/Textures/Objects/Fun/goldbikehorn.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from vgstation at https://github.com/vgstation-coders/vgstation13/commit/125c975f1b3bf9826b37029e9ab5a5f89e975a7e, and modified by github: brainfood1183",
```
Comment (AJCM-git) [inline Resources/Textures/Objects/Fun/goldbikehorn.rsi/meta.json:4]:
```suggestion
  "copyright": "Taken from vgstation at https://github.com/vgstation-coders/vgstation13/commit/125c975f1b3bf9826b37029e9ab5a5f89e975a7e. Modified by brainfood1183 (github) for SS14",
```

File: Resources/Textures/Objects/Specific/Service/vending_machine_restock.rsi/meta.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Original source: https://github.com/tgstation/tgstation/blob/master/icons/obj/vending_restock.dmi @ commit 014c44ef6279beb02a5f3e76824439fa57181c22 - Additions and cleanup by @Vordenburg",
+  "copyright": "Original source: https://github.com/tgstation/tgstation/blob/master/icons/obj/vending_restock.dmi @ commit 014c44ef6279beb02a5f3e76824439fa57181c22 - Additions and cleanup by @Vordenburg, except refill honk which was created by github: brainfood1183",
```
Comment (AJCM-git) [inline Resources/Textures/Objects/Specific/Service/vending_machine_restock.rsi/meta.json:4]:
```suggestion
  "copyright": "Original source: https://github.com/tgstation/tgstation/blob/master/icons/obj/vending_restock.dmi @ commit 014c44ef6279beb02a5f3e76824439fa57181c22 - Additions and cleanup by @Vordenburg. refill_honk made by  brainfood1183 (github) for SS14",
```

File: Resources/Textures/Objects/Storage/happyhonk/clown.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by github: brainfood1183",
```
Comment (AJCM-git) [inline Resources/Textures/Objects/Storage/happyhonk/clown.rsi/meta.json:4]: (same for the rest of copyrights)
```suggestion
  "copyright": "Made by  brainfood1183 (github)",
```

File: Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml [hunk 4]
```diff
@@ -39,3 +39,20 @@
     damage:
       types:
         Blunt: 0
+
+- type: entity
+  parent: BikeHorn
+  id: GoldenBikeHorn
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml:45]: make this worth more than the normal bike horn

File: Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml [hunk 5]
```diff
@@ -39,3 +39,20 @@
...
+  - type: Sprite
+    sprite: Objects/Fun/goldbikehorn.rsi
+    state: icon
+  - type: Item
+    sprite: Objects/Fun/goldbikehorn.rsi
+    size: 5
+  - type: Clothing
+    sprite: Objects/Fun/goldbikehorn.rsi
+    slots: [Belt]
+  - type: Prayable
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml:58]: does the horn still play a sound when you press z with this?

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml (hunk 5 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml:58]: yea it sitll plays a sound, luckily prayable is accessed via the right click menu and not immediate left click.

Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml:58]: yea it still honks with Z

File: Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml (hunk 4 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Fun/bike_horn.yml:45]: i changed the value to 1000, not crazy amount considering rarity.

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/happyhonk.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿- type: vendingMachineInventory
+  id: HappyHonkDispenserInventory
+  startingInventory:
+    HappyHonkRandom: 15
+    HappyHonkMime: 5
```
Comment (metalgearsloth) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/happyhonk.yml:5]: This is way too much, it should be 1/4 of this at max, see https://github.com/space-wizards/space-station-14/issues/12462

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml [hunk 7]
```diff
@@ -368,3 +368,164 @@
...
+  parent: HappyHonkRandom
+  description: A limited mime edition of the happy honk meal.
+  components:
+  - type: Sprite
+    netsync: false
+    sprite: Objects/Storage/happyhonk/mime.rsi
+    state: box
+  - type: Item
+    sprite: Objects/Storage/happyhonk/mime.rsi
+    heldPrefix: box
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml:482]: These don't do anything?

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml [hunk 8]
```diff
@@ -368,3 +368,164 @@
...
+  name: woeful cluwne meal
+  description: Nothing good can come of this.
+  components:
+  - type: Sprite
+    netsync: false
+    sprite: Objects/Storage/happyhonk/cluwne.rsi
+    state: box
+  - type: Item
+    sprite: Objects/Storage/happyhonk/cluwne.rsi
+    heldPrefix: box
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml:530]: As above.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml [hunk 9]
```diff
@@ -368,3 +368,164 @@
     contents:
     - id: FoodDonkpocketDink
       amount: 6
+
+- type: entity
+  id: HappyHonkBase
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml:373]: Base should be the prefix not the suffix, also why isn't happyhonk combined into this?

**Review by metalgearsloth — DISMISSED** [review]
Descriptions need capitalising.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml (hunk 7 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml]: the mime and cluwne meal boxes are currently just different looking boxes for the same thing, to add variety.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml (hunk 8 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml]: mime meal box currently contains same as clown box.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml (hunk 9 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/Containers/box.yml]: i changed this so the base is now clown box reducing the number by 2 entities, but i added a new entity to be used in the random food spawner which i  added it to (includes toy, burger, drink). Hope this is better than previously.

**Comment by EmoGarbage404** [comment]
just gonna comment that, as an obvious happy meal reference, maybe ditch the golden arches on the box? that is mcdonalds whole brand thing and is almost certainly a copyrighted symbol.

........................................................arches
.............................................................|
...........................................................\\|/
image

**Comment by brainfood1183** [comment]
> just gonna comment that, as an obvious happy meal reference, maybe ditch the golden arches on the box? that is mcdonalds whole brand thing and is almost certainly a copyrighted symbol.
> 
> ........................................................arches .............................................................| ...........................................................\|/ image

ye

> just gonna comment that, as an obvious happy meal reference, maybe ditch the golden arches on the box? that is mcdonalds whole brand thing and is almost certainly a copyrighted symbol.
> 
> ........................................................arches .............................................................| ...........................................................\|/ image

Yea, i think a different color other than yellow should be fine but i will change the robust nukie handle from yellow.

**Comment by mirrorcult** [comment]
i love these

**Comment by mirrorcult** [comment]
> maybe i could design handle to look more like HH for happy honk rathe than an M

that is maybe not the most ideal abbreviation but see if it looks fine

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**